From 8007939ce5af7fd2424633d0d1c7625f9e639298 Mon Sep 17 00:00:00 2001
From: Antoine Tenart <antoine.tenart@bootlin.com>
Date: Thu, 9 Aug 2018 10:30:39 +0300
Subject: [PATCH 0536/1051] net: mvpp2: allow to assign a reserved memory
 region for DMA allocations

This patch adds the possibility to reserve a DMA region in the device
tree, and to assign this memory region to the PPv2 device for DMA
allocations. This helps to avoid cases where the default CMA pool is to
small in the kernel, by having a PPv2 specific pool we can control.

Change-Id: I3326471fbda897dcfd22a5b06034b0723d2bd457
Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/59465
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index a3dfea665884..11ba351046a9 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -25,6 +25,7 @@
 #include <linux/of_net.h>
 #include <linux/of_address.h>
 #include <linux/of_device.h>
+#include <linux/of_reserved_mem.h>
 #include <linux/phy.h>
 #include <linux/phylink.h>
 #include <linux/phy/phy.h>
@@ -5389,6 +5390,13 @@ static int mvpp2_probe(struct platform_device *pdev)
 			goto err_dma_mask;
 	}
 
+	/* Assign the reserved memory region to the device for DMA allocations,
+	 * if a memory-region phandle is found.
+	 */
+	if (dev_of_node(&pdev->dev))
+		of_reserved_mem_device_init_by_idx(&pdev->dev,
+						   pdev->dev.of_node, 0);
+
 	/* Initialize network controller */
 	err = mvpp2_init(pdev, priv);
 	if (err < 0) {
-- 
2.17.1

